packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -291,6 +291,20 @@ export class ProjectLifecycleManager {
    *  3. The first browser found.
    */
   async setInitialActiveBrowser () {
+    const browsers = await this.ctx.browser.allBrowsers()
+    const configBrowser = String(this.loadedFullConfig?.browser)
+
+    if (configBrowser !== 'undefined' && (this.ctx.coreData.cliBrowser === 'electron' || !this.ctx.coreData.cliBrowser)) {
+      if (this.ctx.isRunMode) {
+        Object.defineProperty(this.ctx.modeOptions, 'browser', { value: configBrowser })
```

Comment:
Why do we need to `Object.defineProperty`?

Can we just use `setActiveBrowserByNameOrPath`?

Reply:
You need to update packages/config:

https://github.com/cypress-io/cypress/blob/a3e4a8a6067531ffcb1f85598bee9e3d69810fd1/packages/config/src/options.ts#L134-L139

All options need to be defined and validated here, or people won't know about it.

Reply:
> Why do we need to `Object.defineProperty`?
> 
> Can we just use `setActiveBrowserByNameOrPath`?

For Cypress Run Mode, this is the only way I can actually get the browser changed

Reply:
All the options need to be added in here, or we won't get proper error messages. We'll need to figure out what's going on.

I am sure `Object.defineProperty` is not the right solution. I'll try and take a look at this and make some recommendations, give me a day or two.

Reply:
I answered the first part about the option type file in the main thread

Second part I'm still unsure on how to do it the proper way and successfully launch the browser from the config

Reply:
I'm marking this as resolved as we are discussing this in the main thread